File: Design Patterns/Strategy Pattern/Java/com.faculdadevincit.designpatterns.strategypattern/src/SacolaDeCompras.java
public class SacolaDeCompras {
    private IPagamentoStrategy pagamento;

    public IPagamentoStrategy getPagamento() {
        return pagamento;
    }

    public void setPagamento(IPagamentoStrategy pagamento) {
        this.pagamento = pagamento;
    }

    public void realizarPagamento(double valor) {
        if (pagamento == null) {
            System.out.println("Forma de pagamento não definida.");
            return;
        }

        pagamento.pagar(valor);
    }



}

File: Design Patterns/Strategy Pattern/Python/sacola_de_compras.py
from pagamento_strategy import PagamentoStrategy


class SacolaDeCompras:
    pagamento: PagamentoStrategy = None

    def realizar_pagamento(self, valor: float):
        if self.pagamento is None:
            print("Forma de pagamento ainda não definida!")
        else:
            self.pagamento.pagar(valor)